### DISCUSSION TITLE
[MERGED] Merge BeehaveTree and BeehaveRoot?

### DISCUSSION THREAD

**Discussion by bitbrain** [bitbrain/beehave#88] [Polls]
We currently have two nodes that can be used as "root"

- BeehaveTree
- BeehaveRoot

However, the problem with BeehaveTree is that it is not useful unless you actually use a BeehaveRoot (as it contains all the logic). The original thought was that folks can implement their own BeehaveRoot if they really want to but I see this more as a hassle than anything else.

Thoughts?

**Comment by QueenOfSquiggles** [comment]
Whenever I do a behaviour tree implementation, I always have the root be the tree controller as well. Without having looked too deeply into your implementation, I think merging it into one root makes sense. Especially since the user will need to have the root regardless, having effectively two roots feels like it would make the scene tree unnecessarily deep. But I'm also really interested to see what other people think since I'm not the most experienced, just opinionated. :heart:

**Comment by lostptr** [comment]
There's one thing we need to think about. Currently, the `BeehaveTree` is the base class for `BeehaveRoot` and `BeehaveNode` that has the enum for `SUCCESS`, `FAILURE` and `RUNNING`. I don't know if there's a way to make those two classes reference the same enum in gdscript withou inheritance.

**Reply by bitbrain** [reply]
We could define the same enum in each node? Since we return an Integer in tick it does not matter, or am I missing something?

**Comment by bitbrain** [comment]
Raised this PR: https://github.com/bitbrain/beehave/pull/98/files
